src/vs/workbench/services/positronDataExplorer/browser/positronDataExplorerInstance.ts

```diff
@@ -430,6 +430,29 @@ export class PositronDataExplorerInstance extends Disposable implements IPositro
 		notificationHandle.progress.done();
 	}
 
+	/**
+	 * Gets the available code syntaxes.
+	 * @returns A Promise<Array<string>> that resolves with the available code syntaxes.
+	 */
+	async guessCodeSyntax(): Promise<string> {
+		const syntaxes = await this._dataExplorerClientInstance.guessCodeSyntax();
+
+		if (!syntaxes.code_syntax) {
+			return 'No available syntaxes.';
+		}
+		return syntaxes.code_syntax;
+
+	}
+
+	/**
+	 * Translates the data explorer view to code in the desired syntax.
+	 * @returns A Promise<string> that resolves when generated code is returned.
+	 */
+	async translateToCode(desiredSyntax: string): Promise<string> {
```

Comment:
Placing this here for lack of another concrete location: it feels like somewhere there should be an early exit for this entire feature if the backend state has an empty array of supported syntaxes. AFAICT that would be the expected way for the frontend to learn that the backend does not support "export to code".

Reply:
I added a `isSyntaxSupported()` function, as well as a feature support flag (much like the other `supported_features` on the comm) to give some flexibility as we build out different syntax support! We can now early exit when we're converting code, or perhaps even earlier when trying to get a suggestion.